.github/workflows/tests.yml

```diff
@@ -53,71 +56,224 @@ jobs:
-            echo TEST_UPSTREAM=$TEST_UPSTREAM >> $GITHUB_ENV
-            echo "Using development version of dask and distributed"
-          fi
+        run: bash ci/scripts/check_upstream.sh
 
       - name: Install coiled-runtime
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+        run: bash ci/scripts/install_coiled_runtime.sh
+
+      - name: Export environment
+        env:
+          COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        run: bash ci/scripts/export_conda_env.sh
+
+      - name: Build Coiled Software Environment
+        if: ${{ matrix.runtime-version == 'latest' }}
+        env:
+          DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
+          GITHUB_EVENT_NAME: ${{ github.event_name }}
+          GITHUB_EVENT_NUMBER: ${{ github.event.number }}
+          PYTHON_VERSION: ${{ matrix.python-version }}
+          TEST_NAME: runtime
+        run: bash ci/scripts/build_coiled_env.sh
+
+      - name: Run Coiled Runtime Tests
+        continue-on-error: true
+        id: test
+        env:
+            DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
+            AWS_ACCESS_KEY_ID: ${{ secrets.RUNTIME_CI_BOT_AWS_ACCESS_KEY_ID }}
+            AWS_SECRET_ACCESS_KEY: ${{ secrets.RUNTIME_CI_BOT_AWS_SECRET_ACCESS_KEY }}
+            COILED_SOFTWARE_NAME: ${{ env.COILED_SOFTWARE_NAME }}
+            COILED_RUNTIME_VERSION: ${{ matrix.runtime-version }}
         run: |
-          # If testing the latest `coiled-runtime` then install packages defined in `recipe/meta.yaml`
-          # Otherwise, just install directly from the coiled / conda-forge channel
-          if [[ ${{ matrix.runtime-version }} = 'latest' ]]
+          # Ensure we run additional tests when testing the latest coiled-runtime
+          if [[ $COILED_RUNTIME_VERSION = 'latest' ]]
           then
-            python ci/create_latest_runtime_meta.py
-            mamba env update --file latest.yaml
+            export EXTRA_OPTIONS="--run-latest"
           else
-            mamba install -c conda-forge coiled-runtime=${{ matrix.runtime-version }}
+            export EXTRA_OPTIONS=" "
           fi
 
+          python -m pytest $EXTRA_OPTIONS tests/runtime
+
+      - name: Remove Coiled software environment
+        if: ${{ matrix.runtime-version == 'latest' }}
+        env:
+            COILED_SOFTWARE_NAME: ${{ env.COILED_SOFTWARE_NAME }}
+            DASK_COILED__TOKEN: ${{ secrets.COILED_BENCHMARK_BOT_TOKEN }}
+            OUTCOME: ${{ steps.test.outcome }}
+        run: bash ci/scripts/remove_coiled_env.sh
+
+
+  tests_benchmarking:
+    name: Benchmarks ${{ matrix.os }}, Python ${{ matrix.python-version }}, Runtime ${{ matrix.runtime-version }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 120
+    # Required shell entrypoint to have properly activated conda environments
+    defaults:
+      run:
+        shell: bash -l {0}
```

Comment:
I know this predates the changes made in this PR, but we should be able to move this to the top level of the workflow file https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaults. That will avoid us needing to repeat it in each job